src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -0,0 +1,643 @@
+
+    # Register devices and  check response.
+    cbsd_ids = self.assertRegistered(config['registrationRequests'],
+                                     config['conditionalRegistrationData'])
+
+    # Trigger SAS UUT to create a PPA boundary.
+    pal_ids = [record['palId'] for record in config['palRecords']]
+    ppa_creation_request = {
+        "cbsdIds": cbsd_ids,
+        "palIds": pal_ids
+    }
+
+    # Trigger PPA Creation to SAS UUT.
+    ppa_id = self.triggerPpaCreationAndWaitUntilComplete(ppa_creation_request)
+    logging.debug('ppa_id received from SAS UUT:%s', ppa_id)
+
+    # Notify SAS UUT about SAS Harness and trigger Full Activity Dump and
+    # retrieves the PPA Zone record.
+    uut_ppa_zone_data = self.triggerFadGenerationAndRetrievePpaZone(
+        ssl_cert=config['sasTestHarnessCert'],
+        ssl_key=config['sasTestHarnessKey'])
+
+    # Write SAS UUT PPA to output directory of PCR.1 test.
+    # PPA Zone received from SAS UUT in PCR.1 test will be considered as input
+    # for PCR 3,PCR 6 and PCR 7 tests.
+    ppa_zone_data_file_path = os.path.join('testcases', 'output', 'test_WINNF_FT_S_PCR_1',
+                                           'sas_uut_claimed_ppa_boundary.json')
+    writeConfig(ppa_zone_data_file_path, uut_ppa_zone_data['zone'])
+
+    # Asserts the REG-Conditional value doesn't exist in the registrationRequest,
+    # it required to be exist in the registrationRequest data.
+    assertRegConditionalsForPpaRefModel(config['registrationRequests'],
+                                        config['conditionalRegistrationData'])
+
+    # Trigger PPA creation.
+    test_harness_ppa_geometry = ppa.PpaCreationModel(
+        config['registrationRequests'], config['palRecords'])
+
+    # Check if the PPA generated by the SAS UUT is fully contained within the
+    # service area.
+    logging.debug("SAS UUT PPA - retrieved through FAD:%s",
+                  json.dumps(uut_ppa_zone_data, indent=2, sort_keys=False,
+                             separators=(',', ': ')))
+    logging.debug("Reference model PPA - retrieved through PpaCreationModel:%s",
+                  json.dumps(json.loads(test_harness_ppa_geometry), indent=2,
+                             sort_keys=False,
+                             separators=(',', ': ')))
+
+    uut_ppa_geometry = uut_ppa_zone_data['zone']['features'][0]['geometry']
+    self.assertTrue(isPpaWithinServiceArea(config['palRecords'], uut_ppa_geometry),
+                    msg="PPA Zone is not within service area")
+
+    # Check the Equation 8.3.1 in Test Specfification is satisified w.r t
+    # [n.12, R2-PAL-05]. Check the area of the non-overlapping difference between
+    # the maximum PPA boundary created by SAS UUT shall be no more than 10% of the area
+    # of the maximum PPA boundary created by the Reference Model.
+    self.assertTrue(utils.PolygonsAlmostEqual(test_harness_ppa_geometry,
+                                              uut_ppa_geometry))
```

Comment:
I'm assuming that everything above this line is unchanged in this PR. Can you please double-check + confirm?

Edit: I found and commented on a few changed lines. Please confirm that those are the only changes made above this line, or let me know which other ones to look at.

Reply:
Apart   from above two comments rework we have corrected few indentation fix to   maintain the uniformity across this file.     Line No: 221 -  4 space indentation   for line continuation     Line No: 247 -  4 space indentation   for line continuation